From cedafe36afbf9f540616e68233e401cc2d333609 Mon Sep 17 00:00:00 2001
From: Noah Davis <noahadvs@gmail.com>
Date: Wed, 29 Apr 2026 10:08:50 -0400
Subject: [PATCH 2/3] BACKPORT: Reset crop rectangle on canvasRectChanged

A canvasRectChanged signal would likely be caused by another canvas
transformation that would invalidate whatever crop area the user
previously set.

(cherry picked from commit 27b43005c26d8dcd68722705d96cf345891215fe)
---
 src/annotations/CropTool.qml | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/annotations/CropTool.qml b/src/annotations/CropTool.qml
index bdd28da593b5..dd3b8e38e1ab 100644
--- a/src/annotations/CropTool.qml
+++ b/src/annotations/CropTool.qml
@@ -114,6 +114,12 @@ Loader {
                 rect = Utils.rectClipped(rect, baseItem.maxRect())
                 baseItem.setItemRect(selectionItem, rect)
             }
+            function onCanvasRectChanged() {
+                // A canvasRectChanged signal would likely be caused by another
+                // canvas transformation that would invalidate whatever crop
+                // area the user previously set.
+                root.tool.geometry = undefined
+            }
         }
 
         component Overlay: Rectangle {
-- 
2.52.0

